Panic when calling Flush after Close

#### Expected behavior

Flushing on a closed producer should return a proper error.

#### Actual behavior

It panics

#### Steps to reproduce

```Go
producer.Close()
_ = producer.Flush()
```

```
panic: send on closed channel

goroutine 1982 [running]:
github.com/apache/pulsar-client-go/pulsar.(*partitionProducer).FlushWithCtx(0xc0006052c0, {0x3d802d8, 0xc024779d50})
	github.com/apache/pulsar-client-go@v0.13.0/pulsar/producer_partition.go:1485 +0x10c
github.com/apache/pulsar-client-go/pulsar.(*producer).FlushWithCtx(0xc000fd1b80?, {0x3d802d8, 0xc024779d50})
	github.com/apache/pulsar-client-go@v0.13.0/pulsar/producer_impl.go:345 +0xc9
```

#### System configuration
**Pulsar version**: v3.0.5
**Pulsar Go Client**: v13.0

--- a/pulsar/producer_partition.go
+++ b/pulsar/producer_partition.go
@@ -1494,6 +1494,10 @@ func (p *partitionProducer) Flush() error {
 }
 
 func (p *partitionProducer) FlushWithCtx(ctx context.Context) error {
+	if p.getProducerState() != producerReady {
+		return ErrProducerClosed
+	}
+
 	flushReq := &flushRequest{
 		doneCh: make(chan struct{}),
 		err:    nil,

--- a/pulsar/producer_test.go
+++ b/pulsar/producer_test.go
@@ -523,6 +523,21 @@ func TestFlushInPartitionedProducer(t *testing.T) {
 	assert.Equal(t, msgCount, numOfMessages/2)
 }
 
+func TestProducerReturnsErrorOnFlushWhenClosed(t *testing.T) {
+	client, err := NewClient(ClientOptions{URL: serviceURL})
+	assert.NoError(t, err)
+	defer client.Close()
+
+	producer, err := client.CreateProducer(ProducerOptions{Topic: newTopicName()})
+	assert.NoError(t, err)
+	assert.NotNil(t, producer)
+
+	producer.Close()
+
+	err = producer.FlushWithCtx(context.Background())
+	assert.Error(t, err)
+}
+
 func TestRoundRobinRouterPartitionedProducer(t *testing.T) {
 	topicName := "public/default/partition-testRoundRobinRouterPartitionedProducer"
 	numberOfPartitions := 5

